src/main/java/essenmakanan/storage/IngredientStorage.java

```diff
@@ -0,0 +1,101 @@
+package essenmakanan.storage;
+
+import essenmakanan.ingredient.Ingredient;
+import essenmakanan.ingredient.IngredientUnit;
+import essenmakanan.ui.Ui;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class IngredientStorage {
+
+    private final String DATA_PATH = "data/ingredients.txt";
+    private final String DATA_DIRECTORY = "data";
+
+    private ArrayList<Ingredient> ingredientListPlaceholder;
+
+    public IngredientStorage() {
+        ingredientListPlaceholder = new ArrayList<>();
+    }
```

Comment:
when initialising, can handle situations where if file cannot be found or if directory cannot be found